packages/ibm-products/src/components/CreateTearsheet/preview-components/MultiStepWithStepInErrorState.js

```diff
@@ -0,0 +1,146 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState } from 'react';
+import { Button, Column, Grid, TextInput, NumberInput } from '@carbon/react';
+import cx from 'classnames';
+import { pkg } from '../../../settings';
+import { CreateTearsheet } from '../CreateTearsheet';
+import { CreateTearsheetStep } from '../CreateTearsheetStep';
+
+const blockClass = `${pkg.prefix}--tearsheet-create-multi-step`;
+
+export const MultiStepWithStepInErrorState = ({
+  backButtonText,
+  cancelButtonText,
+  className,
+  description,
+  influencerWidth,
+  label,
+  nextButtonText,
+  submitButtonText,
+  title,
+}) => {
+  const [simulatedDelay] = useState(750);
+  const [open, setOpen] = useState(false);
+  const [stepOneTextInputValue, setStepOneTextInputValue] = useState('');
+  const [topicDescriptionValue, setTopicDescriptionValue] = useState('');
+  const [stepTwoTextInputValue, setStepTwoTextInputValue] = useState(1);
+  useState('one-day');
+  const [stepOneIsInvalid, setStepOneIsInvalid] = useState(true);
+  const [stepTwoIsInvalid, setStepTwoIsInvalid] = useState(false);
+
+  const clearCreateData = () => {
+    setStepOneTextInputValue('');
+    setTopicDescriptionValue('');
+    setStepTwoTextInputValue(1);
+    setStepOneIsInvalid(true);
+    setStepTwoIsInvalid(true);
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <style>{`.${blockClass} { opacity: 0 }`};</style>
+      <Button onClick={() => setOpen(!open)}>
+        {open ? 'Close CreateTearsheet' : 'Open CreateTearsheet'}
+      </Button>
+      <CreateTearsheet
+        influencerWidth={influencerWidth}
+        label={label}
+        className={cx(blockClass, className)}
+        submitButtonText={submitButtonText}
+        cancelButtonText={cancelButtonText}
+        backButtonText={backButtonText}
+        nextButtonText={nextButtonText}
+        description={description}
+        title={title}
+        open={open}
+        onClose={clearCreateData}
+        onRequestSubmit={() =>
+          new Promise((resolve) => {
+            setTimeout(() => {
+              clearCreateData();
+              resolve();
+            }, simulatedDelay);
+          })
+        }
+      >
+        <CreateTearsheetStep
+          title="Topic name"
+          fieldsetLegendText="Topic information"
+          disableSubmit={stepOneIsInvalid}
+          subtitle="This is the unique name used to recognize your topic"
+          invalid={stepOneIsInvalid}
+        >
+          <Grid>
+            <Column xlg={8} lg={8} md={8} sm={4}>
+              <TextInput
+                labelText="Topic name"
+                id="tearsheet-multi-step-story-text-input-multi-step-1"
+                value={stepOneTextInputValue}
+                placeholder="Enter topic name"
+                onChange={(event) => {
+                  if (event.target.value.length) {
+                    setStepOneIsInvalid(false);
```

Comment:
One more small request here, if I add a value to the input the invalid state goes away which is expected. But if I empty the text input again it doesn't go back to the invalid state. It would be nice to add `setStepOneIsInvalid(true)` if `!event.target.value.length` otherwise the only time you see the invalid state on the progress indicator is when you initially open the CreateTearsheet.

Reply:
Good catch! Done.